packages/lesswrong/components/form-components/ImageUpload2.tsx

```diff
@@ -0,0 +1,177 @@
+    textTransform: 'none',
+    margin: 5,
+    fontSize: 14,
+    paddingTop: 10,
+    paddingBottom: 10,
+  },
+  imageBackground: {
+    backgroundSize: 'cover',
+    backgroundPosition: 'top right',
+    width: 370,
+    height: 195,
+    display: 'flex',
+  },
+});
+
+const cloudinaryArgsByImageType = {
+  socialPreviewImageId: {
+    minImageHeight: 400,
+    minImageWidth: 700,
+    croppingAspectRatio: 1.91,
+    croppingDefaultSelectionRatio: 1,
+    uploadPreset: cloudinaryUploadPresetSocialPreviewSetting.get(),
+  },
+}
+
+const formPreviewSizeByImageType = {
+  socialPreviewImageId: {
+    width: 306,
+    height: 160
+  },
+}
+
+const ImageUpload2 = ({name, document, updateCurrentValues, clearField, label, croppingAspectRatio, placeholderUrl, classes}: {
+  name: string,
+  document: Object,
+  updateCurrentValues: Function,
+  clearField: Function,
+  label: string,
+  croppingAspectRatio?: number,
+  placeholderUrl?: string,
+  classes: ClassesType
+}) => {
+  const theme = useTheme();
+
+  const setImageInfo = (error, result) => {
+    if (error) {
+      throw new Error(error.statusText)
+    }
+    // currently we ignore all events other than a successful upload -
+    // see list here: https://cloudinary.com/documentation/upload_widget_reference#events
+    if (result.event !== 'success') {
+      return
+    }
+    const imageInfo = result.info
+    if (imageInfo && imageInfo.public_id) {
+      setImageId(imageInfo.public_id)
+      updateCurrentValues({[name]: imageInfo.public_id})
+    } else {
+      //eslint-disable-next-line no-console
+      console.error("Image Upload failed");
+    }
+  }
+
+  const uploadWidget = () => {
+    const cloudinaryArgs = cloudinaryArgsByImageType[name]
+    if (!cloudinaryArgs) throw new Error("Unsupported image upload type")
+    // @ts-ignore
+    cloudinary.openUploadWidget({
+      multiple: false,
+      sources: ['local', 'url', 'camera', 'facebook', 'instagram', 'google_drive'],
+      cropping: true,
+      cloudName: cloudinaryCloudNameSetting.get(),
+      theme: 'minimal',
+      croppingValidateDimensions: true,
+      croppingShowDimensions: true,
+      styles: {
+        palette: {
+            tabIcon: theme.palette.primary.main,
+            link: theme.palette.primary.main,
+            action: theme.palette.primary.main,
+            textDark: "#212121",
+        },
+        fonts: {
+            default: null,
+            "'Merriweather', serif": {
+                url: "https://fonts.googleapis.com/css?family=Merriweather",
+                active: true
+            }
+        }
+      },
+      ...cloudinaryArgs,
+      ...(croppingAspectRatio ? {croppingAspectRatio} : {})
+    }, setImageInfo);
+  }
+  
+  const removeImg = () => {
+    clearField()
+    setImageId(null)
+  }
+
+  const [imageId, setImageId] = useState(() => {
+    if (document && document[name]) {
+      return document[name];
+    }
+    return ''
+  })
```

Comment:
I noticed that this is not properly being set when editing an existing post. It looks like `document.socialPreviewImageId` is `undefined` even if I've manually added a social preview image previously, so maybe we need to update a fragment or something?